packages/components/managed-data-table/src/managed-data-table.js

```diff
@@ -0,0 +1,109 @@
+        isTruncated: !props.displaySettings.disableDisplaySettings
+          ? props.displaySettings.isWrappingText
+          : column.isTruncated,
+      })),
+    [
+      props.columns,
+      props.displaySettings.disableDisplaySettings,
+      props.displaySettings.isWrappingText,
+    ]
+  );
+
+  return (
+    <Spacings.Stack>
+      <DataTableSettings
+        topBar={props.topBar}
+        onSettingsChange={props.onSettingsChange}
+        columnManager={props.columnManager}
+        displaySettings={props.displaySettings}
+      />
+      {React.cloneElement(props.children, {
+        columns,
+        isCondensed: props.displaySettings.isCondensed,
+      })}
+    </Spacings.Stack>
+  );
+};
+
+ManagedDataTable.propTypes = {
+  columns: PropTypes.arrayOf(
+    PropTypes.shape({
+      key: PropTypes.string.isRequired,
+      /* defaults to `auto` */
+      width: PropTypes.string,
+      label: PropTypes.node.isRequired,
+      align: PropTypes.oneOf(['left', 'center', 'right']),
+      onClick: PropTypes.func,
+      /* custom item renderer, specific for items of this column */
+      renderItem: PropTypes.func,
+      isTruncated: PropTypes.bool,
+      isSortable: PropTypes.bool,
+      disableResizing: PropTypes.bool,
+      shouldIgnoreRowClick: PropTypes.bool,
+    })
+  ).isRequired,
+  children: PropTypes.object.isRequired,
+  /* Table settings props */
+  displaySettings: PropTypes.shape({
+    disableDisplaySettings: PropTypes.bool,
+    isCondensed: PropTypes.bool,
+    isWrappingText: requiredIf(
+      PropTypes.bool,
+      (props) => props.displaySettings.disableDisplaySettings
+    ),
+    primaryButton: PropTypes.element,
+    secondaryButton: PropTypes.element,
+  }),
+  columnManager: PropTypes.shape({
+    disableColumnManager: PropTypes.bool,
+    visibleColumnKeys: requiredIf(
+      PropTypes.array,
+      (props) => !props.columnManager.disableColumnManager
+    ),
+    hideableColumns: requiredIf(
+      PropTypes.array,
+      (props) => !props.columnManager.disableColumnManager
+    ),
+    areHiddenColumnsSearchable: PropTypes.bool,
+    searchHiddenColumns: requiredIf(
+      PropTypes.func,
+      (props) =>
+        !props.columnManager.disableColumnManager &&
+        props.columnManager.areHiddenColumnsSearchable
+    ),
+    searchHiddenColumnsPlaceholder: PropTypes.string,
+    primaryButton: PropTypes.element,
+    secondaryButton: PropTypes.element,
+  }),
+  onSettingsChange: requiredIf(
+    PropTypes.func,
+    (props) =>
+      !props.displaySettings.disableColumnManager ||
+      !props.columnManager.disableDisplaySettings
+  ),
+  /* Top bar */
+  topBar: PropTypes.node,
+};
+ManagedDataTable.defaultProps = {
+  isCondensed: false,
+  itemRenderer: (row, column) => row[column.key],
+  displaySettings: {
+    disableDisplaySettings: true,
+  },
+  columnManager: { disableColumnManager: true },
+  isWrappingText: false,
+};
+ManagedDataTable.displayName = 'ManagedDataTable';
```

Comment:
Now that we removed the `<DataTable>` component, I wonder if this is the appropriate name for this package as it might give the wrong impression that this renders a table.
For example we can name this `DataTableMangager`, `TableColumnsManager`, etc.

@jonnybel thoughts on this?

Reply:
Agreed.

I'd go for `DataTableManager`.